pennylane/workflow/qnode.py

```diff
@@ -1113,6 +1107,8 @@ def _execution_component(self, args: tuple, kwargs: dict, override_shots) -> qml
         full_transform_program.set_classical_component(self, args, kwargs)
         _prune_dynamic_transform(full_transform_program, inner_transform_program)
 
+        orig_mcm_config = self.execute_kwargs["mcm_config"]
+        self.execute_kwargs["mcm_config"] = mcm_config

```

Comment:
We need to do this because the inner transform program for legacy devices uses the new `MCMConfig`, and correctness relies on mutations made by `qml.execute` to the new `MCMConfig`.